src/Razor/src/Microsoft.AspNetCore.Razor.Language/Legacy/ParserHelpers.cs

```diff
@@ -19,7 +19,20 @@ internal static class ParserHelpers
             '\u2029' // Paragraph separator
         };
 
-        public static bool IsNewLine(char value) => Array.IndexOf<char>(NewLineCharacters, value) != -1;
```

Comment:
nit: Would it be cleaner to make `NewLineCharacters` a hashset and do a O(1) contains check instead of having two separate code paths?

Reply:
The only reason it's still there is because the test code in the StringTextSnapshot ctor did the following:

delimiterIndex = Content.IndexOfAny(ParserHelpers.NewLineCharacters, start);

I'm going to go ahead and fix that up to not do this. I'd prefer not to do a HashSet as there are hidden costs associated with that and a switch is about as fast as it gets.

---
In reply to: [422416233](https://github.com/dotnet/aspnetcore-tooling/pull/1874#discussion_r422416233) [](ancestors = 422416233)